icu4c/source/i18n/messageformat2_parser.cpp

```diff
@@ -230,16 +254,31 @@ UnicodeSet* initNameChars(UErrorCode& status) {
 
 UnicodeSet* initTextChars(UErrorCode& status) {
     if (U_FAILURE(status)) {
-        return {};
+        return nullptr;
     }
 
+    unisets::gUnicodeSets[unisets::CONTENT] = initContentChars(status);
```

Comment:
I think you can simplfied to
```
    UnicodeSet* content = unisets::gUnicodeSets[unisets::CONTENT] = initContentChars(status);
    UnicodeSet* whitespace = unisets::gUnicodeSets[unisets::WHITESPACE] = initWhitespace(status);
```

Reply:
Fixed in [e2990d2](https://github.com/unicode-org/icu/pull/3306/commits/e2990d2b9ccf2f31832348fa29d0f36ff4c3002f)